consensus/istanbul/validator/default.go

```diff
@@ -251,16 +291,27 @@ func (valSet *defaultSet) AsData() *istanbul.ValidatorSetData {
 	}
 }
 
+func (valSet *defaultSet) AsDataWithCache() *istanbul.ValidatorSetDataWithCache {
+	valSet.validatorMu.RLock()
+	defer valSet.validatorMu.RUnlock()
+	return &istanbul.ValidatorSetDataWithCache{
+		Validators: MapValidatorsToDataWithCache(valSet.validators),
+		Randomness: valSet.randomness,
+	}
+}
+
 // JSON Encoding -----------------------------------------------------------------------
 
-func (val *defaultSet) MarshalJSON() ([]byte, error) { return json.Marshal(val.AsData()) }
+func (val *defaultSet) MarshalJSON() ([]byte, error) {
```

Comment:
do we need tho change this method? it's not used by snapshot, and it breaks simmetry. If i marshall as json it's on thing, if a marshall with RLP it's another